### DISCUSSION TITLE
editing & reloading of config.ini

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#262] [Ideas]
**Editing:**

There's a number of options (perhaps more to come ;-)  that users need to customize manually by editing config.ini. However, trying to locate and open the config.ini file can be a challenge since we don't do this regularly, due to it's rather obscured location, different path across platforms, etc. 

I notice the file path of config.ini is shown in the Save Preference dialog. Unfortunately, on Windows (why?) I can't select the path at all. I can select the path on Linux but it's still not as straightforward. Perhaps we can make the path in the dialog a link that we can click and open or, maybe easier to achieve, add a button to let user open the file?

**Reloading:**

One other issue is that we have to restart nedit/NG to pick up the new changes. Maybe nedit can check and reload the setting if the config file has been modified, just like the documents in nedit.

**Comment by eteran** [comment]
I have considered adding a new menu item under Help that is "open configuration file" :-)

**Comment by tksoh** [comment]
That would work too :-)

**Comment by eteran** [comment]
As for dynamically picking up the changes... that's probably a 6.1/6.2 feature, if at all. I think it's pretty common for editors to need a restart if you manually edit the config files outside of the UI.

**Reply by tksoh** [reply]
sure

**Comment by eteran** [comment]
@tksoh I've added a new menu item to the Help menu in https://github.com/eteran/nedit-ng/pull/273 .

If you think this suits your needs nicely, I can merge it :-)

**Comment by tksoh** [comment]
That's great. WIll try it out soon. Thanks

**Comment by tksoh** [comment]
Although I am very slightly unsure it belongs under the Help menu, but it works great! Even better than I originally suggested, as we can just open anything inside the folder by DnD, so we can quick locate, view and share the other yaml files -- all without having to add countless menu to open this yaml and that yaml.

With this, I now really feel we should also add the menu to open the folder containing the files being edited. IMO, it works better than trying to open file(s) using the open dialog, since I can keep the folder opened at all time.

**Reply by eteran** [reply]
Glad it works well for you!

I am definitely considering where it would be best located. Help, probably won't be it's final location.

Regarding your last sentence, i to make sure I understand it, are you suggesting that for whatever file is currently open, we have a shortcut/menu to open a file browser to th directory that the file is in?

**Reply by eteran** [reply]
@tksoh BTW, I think I'll be adding an "Open Parent Directory" to the tab context menu. I think that's a nice simple and clean place to add the functionality.

**Reply by tksoh** [reply]
That'd be great. Perhaps we should always at the menu to copy the path? It beats manually copying from the statistic line, and also to prepare for the future work on status bar.

**Reply by eteran** [reply]
Yea, I think that additive features to context menus is pretty fair game as long as it doesn't get out of hand.

**Reply by tksoh** [reply]
Absolutely agreed. With all due respect, the tab context menu on Notepad++ is a good example of what we should not do :-P
